src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -216,6 +218,11 @@ protected INodeSelector GetNodeSelector(Topology initialTopology)
             }
         }
 
+        public void RetireSession(long sessionId)
+        {
+            _nodeSelectorBySessionId.TryRemove(sessionId, out _);
```

Comment:
And now not calling this method will produce a memory leak?
It also means that we are going to hold on to sessions for a lot longer.